inline-python-macros/src/lib.rs

```diff
@@ -21,17 +18,34 @@ use embed_python::EmbedPython;
 mod meta;
 use self::meta::{Meta, NameValue};
 
-#[proc_macro]
-pub fn python(input: TokenStream1) -> TokenStream1 {
+/// Create a syn::Error with an optional span, and a format string with arguments.
+///
+/// If no span is given, it defaults to Span::call_site().
+///
+/// For example:
+/// ```no_compile
+/// error!(token.span(), "foo: {}", "bar"); // With span.
+/// error!("foo: {}", "bar"); // Without span.
+/// ```
+macro_rules! error {
+	($format:literal) => ( syn::Error::new(Span::call_site(), format!($format)) );
+	($format:literal, $($tokens:tt)*) => ( syn::Error::new(Span::call_site(), format!($format, $($tokens)*)) );
+	($span:expr, $format:literal) => ( syn::Error::new($span, format!($format)) );
+	($span:expr, $format:literal, $($tokens:tt)*) => ( syn::Error::new($span, format!($format, $($tokens)*)) );
```

Comment:
Can't you combine the cases with and without `, $($tokens:tt)*`?

Reply:
Yes, simplified :)